contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -319,27 +350,15 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
 }
 
 impl InterchainTokenService {
-    fn pay_gas_and_call_contract(
+    fn send_to_hub(
         env: &Env,
-        caller: Address,
-        destination_chain: String,
-        message: Message,
+        spender: Address,
+        payload: Bytes,
```

Comment:
```suggestion
        payload: HubMessage,
```
Take a typed message instead of arbitrary bytes for safety

Reply:
good point, updated